app/cdap/components/ResourceCenterEntity/PullPipelineWizard.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import Alert from 'components/shared/Alert';
+import React from 'react';
+import { Provider } from 'react-redux';
+import SourceControlManagementSyncStore from 'components/SourceControlManagement/store';
+import { RemotePipelineListView } from 'components/SourceControlManagement/RemotePipelineListView';
+import StandardModal from 'components/shared/StandardModal';
+import T from 'i18n-react';
+import { IAction } from 'services/redux-helpers';
+
+export const reducer = (state, action) => {
+  switch (action.type) {
+    case 'TOGGLE_MODAL':
+      return {
+        ...state,
+        isOpen: !state.isOpen,
+      };
+    case 'SET_ERROR':
+      return {
+        ...state,
+        error: action.payload.error,
+      };
+    case 'SET_LOADING':
+      return {
+        ...state,
+        loading: action.payload.loading,
+      };
+  }
+};
+
+export const defaultState = {
+  isOpen: false,
+  error: null,
+  loading: false,
+};
+
+interface IPullPipelineWizardProps {
+  isOpen: boolean;
+  error: string;
+  dispatch: React.Dispatch<IAction>;
```

Comment:
type this with the reducer above so you can only do SET_LOADING, SET_ERROR and TOGGLE_MODAL -- the type you imported has payload any and doesn't match the strings - you can use that IAction to create the correct types - ping me later if you need help on this

Reply:
Did some change. Let me know if this is what you are looking for